lib/spack/spack/spec.py

```diff
@@ -683,7 +778,7 @@ def to_node_dict(self):
 
     def to_yaml(self, stream=None):
         node_list = []
-        for s in self.traverse(order='pre'):
+        for s in self.traverse(order='pre', deptype=(spack.DEP_INSTALL, spack.DEP_RUN)):
```

Comment:
I think we still want to write out the build deps as part of the provenance for each build.  So this ought to generate YAML that includes them but doesn't hash them.  And they should be marked as build deps...